src/node_buffer.cc

```diff
@@ -1489,6 +1489,78 @@ uint32_t FastWriteString(Local<Value> receiver,
 
 static v8::CFunction fast_write_string(v8::CFunction::Make(FastWriteString));
 
+uint32_t FastWriteStringUTF8(
+    Local<Value> receiver,
+    const v8::FastApiTypedArray<uint8_t>& dst,
+    const v8::FastOneByteString& src,
+    uint32_t offset,
+    uint32_t max_length,
+    // NOLINTNEXTLINE(runtime/references) This is V8 api.
+    v8::FastApiCallbackOptions& options) {
+  uint8_t* dst_data;
+  CHECK(dst.getStorageIfAligned(&dst_data));
+  CHECK(offset <= dst.length());
+  CHECK(dst.length() - offset <= std::numeric_limits<uint32_t>::max());
+
+  const auto size = std::min(
+      {static_cast<uint32_t>(dst.length() - offset), max_length, src.length});
+
+  auto dst_ptr = dst_data + offset;
+  auto src_ptr = reinterpret_cast<const uint8_t*>(src.data);
+
+  // TODO(ronag): Use simdutf for this.
+  // Produces quite good assembly with:
+  // gcc -O3 -mavx -mavx2 -mavx512f
+
+  size_t i = 0;
+
+  // 256 bit
+  for (; i + 32 <= size; i += 32) {
+    uint64_t f = 0;
+    uint64_t v = 0;
+    memcpy(&v, src_ptr + i, 8);
+    f |= v;
+    memcpy(&v, src_ptr + i + 8, 8);
+    f |= v;
+    memcpy(&v, src_ptr + i + 16, 8);
+    f |= v;
+    memcpy(&v, src_ptr + i + 24, 8);
+    f |= v;
+    if ((f & UINT64_C(0x8080808080808080)) != 0) {
+      options.fallback = true;
```

Comment:
I won't block because this fixes a regression but I'm disappointed to see `option.fallback` come back after me and the V8 team took the time to remove it from the codebase with the V8 12.8 update.

Reply:
The changes in this PR also go against the principles of the V8 Fast API (as far as I understand them).
The point is to make the JS<->C++ interaction fast, not to implement complicated code that is only called when the fast path is hit.
All the optimizations that you did in this PR:
- Could also benefit the "slow" version of the method
- Are certainly not tested by the regression test you added, given all the new fallback paths.

Reply:
> I won't block because this fixes a regression but I'm disappointed to see option.fallback come back after me and the V8 team took the time to remove it from the codebase with the V8 12.8 update.

I don't know how to implement this without the fallback option.

An alternative is to just not use fast api for utf8. Would you prefer that? I can prepare a PR for that instead.

Reply:
> Could also benefit the "slow" version of the method

I'm happy to continue iterating. I think we should probably wait for simdutf to implement it and then refactor.

> Are certainly not tested by the regression test you added, given all the new fallback paths.

Ditto with simduft.

Reply:
I'm not sure what's blocking you so it's difficult to help.
Let's take a step back: the "slow" version ends up calling `StringBytes::Write`, which calls `str->WriteUtf8`.
Is that what you want to avoid doing in the "fast" version?

Reply:
```Write(Isolate* isolate,
                          char* buf,
                          size_t buflen,
                          Local<Value> val,
                          enum encoding encoding)```

I don't know how to get the `Local<Value>`

Reply:
```c++
HandleScope scope(options.isolate);
// Create the handle
// Pass the handle
```

Reply:
Sorry. I don't understand how that helps?

Reply:
You're saying you don't know how to passe the `Local<Value>`, since that's obvious (you just pass it like any other variable), I assumed you don't know how to create it in this context. To create a `Local`, you need a `HandleScope`, which you can create using the isolate that V8 passes in the options.

Reply:
So I have a OneByteString reference. StringWrite does not that take.

Reply:
`String::NewFromOneByte(options.isolate, src_ptr, NewStringType::kNormal, src.length)`

Reply:
I'll have another go tomorrow. Not sure how urgent this is to land.